torchtune/models/clip/_model_builders.py

```diff
@@ -1,4 +1,63 @@
-from torchtune.models.clip._transforms import CLIPImageTransform
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from pathlib import Path
+
+from torchtune.models.clip._text_encoder import CLIPTextEncoder
+from torchtune.models.clip._tokenizer import CLIPTokenizer
+from torchtune.models.clip._transform import CLIPImageTransform
+from torchtune.utils._download import TORCHTUNE_LOCAL_CACHE_FOLDER, download_file
+
+CLIP_VOCAB_URL = "https://github.com/openai/CLIP/raw/refs/heads/main/clip/bpe_simple_vocab_16e6.txt.gz"
+
+
+def clip_tokenizer(
```

Comment:
Take a look at some of the other tokenizer builder signatures. We've tried to standardize around support prompt templates from users here, though it might not makes sense as CLIP doesn't really use these.

Reply:
yeah I figured that didn't make sense since no one is going to be doing chat templates, etc. with CLIP. 

But I could include it as an option since maybe there's some other kind of template that would be used. I could see users wanting to do some simple templating like e.g. dreambooth's "a {unique_identifier} {class}".

But I don't think we should structure it like other tokenizers/templates that support a list of messages, because no one will want that for CLIP, especially with its 77 context length